scripts/setup/nrfconnect/update_ncs.py

```diff
@@ -0,0 +1,129 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import os
+import subprocess
+import sys
+
+def get_repository_commit_sha(repository_location):
+<<<<<<< HEAD
+    command = f"git -C {repository_location} rev-parse HEAD"
+    process = subprocess.run(command.split(), check=True, stdout=subprocess.PIPE)
+=======
+    command = ['git', '-C', repository_location, 'rev-parse', 'HEAD']
+    process = subprocess.run(command, check=True, stdout=subprocess.PIPE)
+>>>>>>> b05e9d3f003005555d085e4a14d0c1f5a6422f43
+    return process.stdout.decode('ascii').strip()
+
+def update_ncs(repository_location, version):
+    # Fetch sdk-nrf to the desired version.
+<<<<<<< HEAD
+    command = f"git -C {repository_location} fetch origin"
+    subprocess.run(command.split(), check=True)
+
+    # Checkout sdk-nrf to the desired version.
+    command = f"git -C {repository_location} checkout {version}"
+    subprocess.run(command.split(), check=True)
+
+    # Call west update command to update all projects and submodules used by sdk-nrf.
+    command = "west update -r"
+    subprocess.run(command.split(), check=True)
+=======
+    command = ['git', '-C', repository_location, 'fetch']
+    subprocess.run(command, check=True)
+
+    # Checkout sdk-nrf to the desired version.
+    command = ['git', '-C', repository_location, 'checkout', version]
+    subprocess.run(command, check=True)
+
+    # Call west update command to update all projects and submodules used by sdk-nrf.
+    command = ['west', 'update', '-r']
+    subprocess.run(command, check=True)
+>>>>>>> b05e9d3f003005555d085e4a14d0c1f5a6422f43
+
+def get_ncs_recommended_version():
+    chip_root = os.path.abspath(os.path.join(os.path.dirname(__file__), os.path.normpath('../../..')))
+
+    # Read recommended version saved in the .nrfconnect-recommended-version file.
+    try:
+        with open(os.path.join(chip_root, 'config/nrfconnect/.nrfconnect-recommended-version'), 'r') as f:
+            return f.readline().strip()
+    except:
+        raise RuntimeError("Encountered problem when trying to read .nrfconnect-recommended-version file.")
+
+def print_messages(messages : list, yellow_text :  bool):
+    # Add colour formatting if yellow text was set
+    if yellow_text:
+        messages = [f"\33[33m{message}\x1b[0m" for message in messages]
+
+    for message in messages:
+        print(message)
+
+def print_check_version_warning_message(current_version, recommended_version):
+    current_version_message = f"WARNING: Your current NCS version ({current_version})"
+    recommended_version_message = f"differs from the recommended ({recommended_version})."
```

Comment:
I'd use `latest_stable_version_message` and remove `recommended` whatsoever.

Reply:
I replaced `versions` with `revisions`, but to the rest of suggestions I replied in the comment about github workflows.

Reply:
We should not use "latest_stable" prefix as it won't be true one. Here we provide recommended compatible revision of the NCS. It might not be "stable" or latest stable. And the word "recommended" means, that basically it is something which we tested against, but newer versions (or sometimes even older) might be workable as well (in case some clients would like to use a different one).

I think we don't need to introduce a compatibility matrix since each Project CHIP revision keeps recommended version of NCS, so it's just a matter of cloning certain CHIP version that client wants.